cdk/index.ts

```diff
@@ -1,19 +1,55 @@
+import * as cognito from "@aws-cdk/aws-cognito";
+import * as ssm from "@aws-cdk/aws-ssm";
 import * as cdk from "@aws-cdk/core";
 import "source-map-support/register";
 import { StaticSite } from "./static-website";
 
+interface StaticSiteProps extends cdk.StackProps {
+  environmentId: string;
+}
+
 class StaticSiteStack extends cdk.Stack {
-  constructor(parent: cdk.App, name: string, props: cdk.StackProps) {
+  constructor(parent: cdk.App, name: string, props: StaticSiteProps) {
     super(parent, name, props);
 
```

Comment:
Worth having any error handling here to fail gently if the IDP can't be found?

Reply:
I think probably no. The CDK should hard bail if the `fromStringParameterName`  lookup fails. It won't necessarily be a super friendly message but I don't generally expect humans to be deploying this stack and I do expect that those who do (or those who review pipeline logs) will be able to know what the CDK errors mean. And I don't mean this in a way of like "if you don't understand the raw gobbledygook that the CDK throws at you, you're not allowed to deploy" but just, "I think the the full trace and error message is probably more informative than something I would write".

This does bring up another point. We want to support building out various front-end "environments" for sandboxing but it may be incorrect to expect them all to build their own backend stack. Right now, it's sort of built such that they can't be shared. We may want to reconsider this in the future. But in the interest of moving things forward, I think we can add that feature later. We've lived with only one front-end stack for a very long time; most sandbox testing happens locally.

Reply:
That latter point is a good one.  Probably some utility to be had in pairing multiple front-end deployments to share backend stacks.

Reply:
Totally reasonable. I'm actually inclined to let the UI devs deploy their own sandboxes as needed because it's really pretty simple. I think this is definitely good for the first cut.

Reply:
@jasonburkert-ccpo and @jeffsegal-ccpo I'll put together a small ticket to build out that functionality in an upcoming sprint to resolve this thread. Maybe also try to get some time next week to talk about "what cool stuff can we do with the pipeline".